client/components/social-buttons/google.js

```diff
@@ -191,6 +206,8 @@ class GoogleLoginButton extends Component {
 			this.state.isDisabled || this.props.isFormDisabled || this.state.error
 		);
 
+		const isLoading = this.state.isClicked && ! this.state.initialized && ! this.state.error;
```

Comment:
If the `loadDependency().catch()` handler reset the `clickEvent` back to `null`, then `isLoading` is equivalent to `!! this.state.clickEvent`. There is a deferred `clickEvent` if and only if the API is loading.

I would also carefully look at the "error" code path. If the API load fails, what happens to the saved `clickEvent`? Is it handled and how? What happens when I repeatedly click on the button after the API load fails?